### ISSUE TITLE
Reproducing results in the paper

### ISSUE THREAD

**Issue by ngailapdi** [szymanowiczs/splatter-image#11]
Hi, thank you for the great work.

I followed the default parameters in the config file and trained on SRN chairs single-view. However, I did not get the same results as in the paper at the end.
```
PSNR_novel: 18.43
LPIPS_novel: 0.10
SSIM_novel: 0.89
```
I wonder if the hyperparameters used to train your model differ from the default in the config file? Can you please provide the hyperparameters that you used?

**Comment by mengxuyiGit** [comment]
and my lpips loss remains unchanged during training, how about your training loss curves? Many thanks!
<img width="1440" alt="image" src="https://github.com/szymanowiczs/splatter-image/assets/74002878/7c54262b-aca8-4483-9788-a351cb8c6363">

**Comment by ngailapdi** [comment]
> and my lpips loss remains unchanged during training, how about your training loss curves? Many thanks! <img alt="image" width="1440" src="https://private-user-images.githubusercontent.com/74002878/295380298-7c54262b-aca8-4483-9788-a351cb8c6363.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4NTE2MTUsIm5iZiI6MTcwNDg1MTMxNSwicGF0aCI6Ii83NDAwMjg3OC8yOTUzODAyOTgtN2M1NDI2MmItYWNhOC00NDgzLTk3ODgtYTM1MWNiOGM2MzYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDAxNDgzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlNzllNTgxZTdhNzQ1NjQ1YTE4NmVkYTMyZGJjMmMzMzViYWRkNjY5MzM1NGViN2UzNWY2MTI3MzE4ODg3ZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.t67QF68NK4NATjAVefMHurvNUSdBku1nPXaOMnggY7g">

Hi, lpips loss does not get applied until 800K steps so you won't see any change until then

**Comment by yuchenlichuck** [comment]
```
Setting up [LPIPS] perceptual loss: trunk [vgg], v[0.1], spatial [off] [10/01 17:01:14]
/home/dubaiprince/miniconda3/envs/gsgen/lib/python3.9/site-packages/torchvision/models/_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
/home/dubaiprince/miniconda3/envs/gsgen/lib/python3.9/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=VGG16_Weights.IMAGENET1K_V1`. You can also use `weights=VGG16_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
Loading model from: /home/dubaiprince/miniconda3/envs/gsgen/lib/python3.9/site-packages/lpips/weights/v0.1/vgg.pth [10/01 17:01:15]
Beginning training [10/01 17:01:15
```

**Comment by yuchenlichuck** [comment]
My code is stuck here

**Comment by szymanowiczs** [comment]
@ngailapdi can you share what command you used to train the model? how many iterations did you train for? what do your validation PSNR curves look like? Do the visualisations look reasonable?

**Comment by Kevin-Miao** [comment]
> Hi! That's amazing you can train the network! Have you encountered any problem of being killed when training? Thanks! <img alt="image" width="285" src="https://private-user-images.githubusercontent.com/74002878/295376396-8f9f2afa-f3b1-427b-a373-c4c62a40d204.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUxODU5NTYsIm5iZiI6MTcwNTE4NTY1NiwicGF0aCI6Ii83NDAwMjg3OC8yOTUzNzYzOTYtOGY5ZjJhZmEtZjNiMS00MjdiLWEzNzMtYzRjNjJhNDBkMjA0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEzVDIyNDA1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkMzdkOTczNzBkZDQ2ODQ1ZjczZjEzOTU0MmQ5M2Y3YjFmMDMyMzAyYjA2MDk3YTlmNTUzNWVkNTUyNjA4ZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wCTbfwW0urB_agHS_Wyb7q84s0WwudPkbddwWeCFxCo">

I was experiencing similar issues, whereby I thought some memory leak was hoarding my CPU memory. I found out that it is due to the implementation of the dataset class in `srn.py` -- specifically, the `load_example_id(...)` function. I changed the script to not cache specific entries. I guess more CPU memory could also relieve this, but probably wouldn't work with gigantic datasets.

**Comment by ngailapdi** [comment]
> At a glance, the parameters in the provided config should take you to the same results as in the paper. @ngailapdi can you share what command you used to train the model? how many iterations did you train for? what do your validation PSNR curves look like? Do the visualisations look reasonable? Can you share your config.yaml file that the code output?

Hi, thank you for your response. This is the command I use to train the model on SRN chair
`python train_network.py +dataset=[chairs]`
I trained for 1,000,000 iterations, here is the PSNR novel curve

![image](https://github.com/szymanowiczs/splatter-image/assets/17942095/466c36db-ed78-46db-ba02-73dbcafd6d0a)

and here is the visualization at the last iteration

![image](https://github.com/szymanowiczs/splatter-image/assets/17942095/ef3d09d7-abe5-4d0f-b98e-017ec15819af)

and here is the output `config.yaml`. I just took a look at the appendix, it seems like the learning rate in the default config is different from the one reported in the appendix (5e-5 vs 2e-4), I'm not sure with 1M iterations if this will make a big difference.

```
wandb_version: 1

wandb:
  desc: null
  value:
    project: gs_pred
cam_embd:
  desc: null
  value:
    embedding: null
    encode_embedding: null
    dimension: 0
    method: null
general:
  desc: null
  value:
    device: 0
    random_seed: 0
data:
  desc: null
  value:
    training_resolution: 128
    fov: 51.98948897809546
    subset: -1
    input_images: 1
    origin_distances: false
    rescale_to_cars: false
    set_to_pm_1_range: true
    transform_shs: true
    depth_rendering: false
    znear: 1.25
    zfar: 2.75
    category: chairs
    white_background: true
opt:
  desc: null
  value:
    iterations: 1000001
    base_lr: 5.0e-05
    batch_size: 8
    betas:
    - 0.9
    - 0.999
    loss: l2
    imgs_per_obj: 4
    ema:
      use: true
      update_every: 10
      update_after_step: 100
      beta: 0.9999
    lambda_lpips: 0.01
    start_lpips_after: 800001
    pretrained_ckpt: null
    step_lr_at: 800001
diffusion:
  desc: null
  value:
    channel_mult_noise: 0
model:
  desc: null
  value:
    max_sh_degree: 1
    inverted_x: false
    inverted_y: true
    name: SingleUNet
    out_channels: 64
    opacity_scale: 0.001
    opacity_bias: -2.0
    scale_bias: 0.02
    scale_scale: 0.001
    xyz_scale: 1.0e-06
    xyz_bias: 0.0
    depth_scale: 1.0
    depth_bias: 0.0
    network_without_offset: false
    network_with_offset: true
    attention_resolutions:
    - 16
    num_blocks: 4
    cross_view_attention: true
    base_dim: 128
    isotropic: false
    anneal_opacity: true
logging:
  desc: null
  value:
    ckpt_iterations: 1000
    val_log: 10000
    loss_log: 10
    loop_log: 10000
    render_log: 10000
_wandb:
  desc: null
  value:
    python_version: 3.10.13
    cli_version: 0.16.1
    framework: huggingface
    huggingface_version: 4.33.3
    is_jupyter_run: false
    is_kaggle_kernel: false
    start_time: 1703174534.939141
    t:
      1:
      - 1
      - 11
      - 41
      - 49
      - 50
      - 55
      - 105
      2:
      - 1
      - 11
      - 41
      - 49
      - 50
      - 55
      - 105
      3:
      - 2
      - 5
      - 14
      - 16
      - 23
      4: 3.10.13
      5: 0.16.1
      6: 4.33.3
      8:
      - 5
      13: linux-x86_64
```

**Comment by szymanowiczs** [comment]
OK thanks for letting me know. at a glance the hyper parameters look fine. No camera embedding is needed. I'll dig in on my side and get back to you.

**Comment by Kevin-Miao** [comment]
@szymanowiczs I am trying to understand the parameters used in the `configs/dataset` folder. How did you precisely identify `zfar` and `znear`? How did you find them impacting your results?

**Comment by szymanowiczs** [comment]
Hi @ngailapdi can you pull the latest version of the code and try again? I ran training with the released code and got results like these: blue is cars, grey is chairs. So within 100k iterations you should expect to get novel PSNR on the validation set of 22.8 for cars and 22.0 for chairs. Let me know if it works for you.

![Screenshot 2024-01-22 at 09 21 46](https://github.com/szymanowiczs/splatter-image/assets/23418488/e72818f7-492b-4bb2-ae80-a6883b43dbf6)

**Comment by yuchenlichuck** [comment]
> > My code is stuck here
> 
> @yuchenlichuck is the code actually stuck or are just not seeing any print statements? if you're not seeing any print statements that's fine - the code doesn't print output until 10k iterations in when it begins validation.

Yes, you are right, it fixed now

**Comment by szymanowiczs** [comment]
Great, I'm closing the issue for now - if you encounter further issues with reproducing results on ShapeNet classes feel free to reopen it.

**Comment by SJTUwxz** [comment]
Hi! Thanks again for the great work!

 I'm reopening this issue again as I ran the training using default parameters twice, but neither of which gets the same result on SRN-cars and SRN-chairs test dataset as reported in the paper (especially PSNR and LPIPS metrics). Please see the results below. Evaluating using the released checkpoint gets very similar results though.
<img width="702" alt="result-replication" src="https://github.com/szymanowiczs/splatter-image/assets/10277398/fe49d7d9-4ea0-4665-b9f1-e0b5ecce654f">

I'd really appreciate it if you could provide some insights about this issue. Thank you in advance!

**Comment by szymanowiczs** [comment]
Thanks for highlighting. I'll look into this - my intuition is that this could be due to stochasticity in training, so I'll see if there is a way to fix it to make sure results can be reproduced exactly.

**Comment by SJTUwxz** [comment]
Thanks for your response! I will look forward to that!
